### PR TITLE
Whitelist support creation

### PR DESCRIPTION
**Pull request by marcos-iov**
- Use WhitelistResponseCode in Bridge class
- Add network parameters to WhitelistConstants. Remove network parameters from WhitelistSupport, get the value from WhitelistConstants
- Update BridgeSupportFactory to build WhitelistSupport
- Include WhitelistConstants tests

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java [hunk 1]
```diff
@@ -93,25 +104,42 @@ public BridgeSupport newInstance(Repository repository, Block executionBlock,
...
+
         return new FeePerKbSupportImpl(feePerKbConstants, feePerKbStorageProvider);
     }
+
+    private WhitelistSupport getWhitelistSupportInstance(StorageAccessor bridgeStorageAccessor, ActivationConfig.ForBlock activations) {
+        WhitelistConstants whitelistConstants = bridgeConstants.getWhitelistConstants();
+        WhitelistStorageProvider whitelistStorageProvider = new WhitelistStorageProviderImpl(
+            whitelistConstants.getBtcParams(),
+            activations,
+            bridgeStorageAccessor
```
Comment (julia-zack) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java:136]: i just noticed something when seeing this.
When doing the FederationSupport refactor, we had [this](https://github.com/rsksmart/rskj/pull/2330#discussion_r1603951907) discussion about the activations and network parameters being part of the class.

This situation is different because the _activations_ are not part of the `WhitelistSupportImpl`, but the _networkParameters_ are. So probably they should be removed from the `WhitelistStorageProvider` class.

About the activations, I think we should take advantage of the fact that the `BridgeSupport` already has an _activations_ field, and create the supports with it.
That way, all the storage providers would only need the bridge storage accessor to be created, standardizing it.
I find this approach cleaner.

wdyt guys? @marcos-iov @nathanieliov @jeremy-then @apancorb

File: rskj-core/src/main/java/co/rsk/peg/whitelist/constants/WhitelistConstants.java [hunk 2]
```diff
@@ -1,11 +1,16 @@
 package co.rsk.peg.whitelist.constants;
 
+import co.rsk.bitcoinj.core.NetworkParameters;
 import co.rsk.peg.vote.AddressBasedAuthorizer;
 
 public class WhitelistConstants {
-
+    protected NetworkParameters btcParams;
```
Comment (wilmerrootstock) [inline rskj-core/src/main/java/co/rsk/peg/whitelist/constants/WhitelistConstants.java:7]: Is it necessary to include `btcParams` as part of the Whitelist objects?  I noticed this attribute is already in BridgeConstants, and putting it in other places could affect the data modeling. If we see that data model to get a good normalization we have to avoid spreading the same information/attribute in different objects.

File: rskj-core/src/main/java/co/rsk/peg/Bridge.java [hunk 3]
```diff
@@ -1055,20 +1052,20 @@ public long getLockWhitelistEntryByAddress(Object[] args) {
         try {
             addressBase58 = (String) args[0];
         } catch (Exception e) {
-            logger.warn("Exception in getLockWhitelistEntryByAddress", e);
-            return LOCK_WHITELIST_INVALID_ADDRESS_FORMAT_ERROR_CODE;
+            logger.warn("[getLockWhitelistEntryByAddress] Exception in getLockWhitelistEntryByAddress", e);
```
Comment (wilmerrootstock) [inline rskj-core/src/main/java/co/rsk/peg/Bridge.java:1055]: Awesome! Looks much better with the pattern: logger.type("[method]"...). 
WDYT if we add a variable with some information and remove "Exception in getLockWhitelistEntryByAddress" (it seems redundant with the name of the method):

`logger.warn("[removeLockWhitelistAddress] {}", e.getMessage(), e);`

File: rskj-core/src/main/java/co/rsk/peg/whitelist/constants/WhitelistConstants.java (hunk 2 above)
Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/whitelist/constants/WhitelistConstants.java:7]: Makes life easier :) I can get the network params from the constants instead of passing the NetowrkParam object around.

I agree that it adds some duplication, but the value is also tied to the class name (MainnetConstants => MainnetParams). I'm open for discussing it, there's room for improvement in many thing s we've done so far definitely

File: rskj-core/src/main/java/co/rsk/peg/Bridge.java (hunk 3 above)
Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/Bridge.java]: Done, thanks!

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java (hunk 1 above)
Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java]: Thanks for the idea! Just implemented it, let me know what you think

File: rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java [hunk 4]
```diff
@@ -50,9 +58,9 @@ public LockWhitelistEntry getLockWhitelistEntryByIndex(int index) {
     @Override
     public LockWhitelistEntry getLockWhitelistEntryByAddress(String addressBase58) {
         try {
-            Address address = Address.fromBase58(networkParameters, addressBase58);
+            Address address = Address.fromBase58(whitelistConstants.getBtcParams(), addressBase58);
```
Comment (julia-zack) [inline rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java:61]: why this change?

File: rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java [hunk 5]
```diff
@@ -62,7 +70,7 @@ public LockWhitelistEntry getLockWhitelistEntryByAddress(String addressBase58) {
     @Override
     public int addOneOffLockWhitelistAddress(Transaction tx, String addressBase58, BigInteger maxTransferValue) {
         try {
-            Address address = Address.fromBase58(networkParameters, addressBase58);
+            Address address = Address.fromBase58(whitelistConstants.getBtcParams(), addressBase58);
```
Comment (julia-zack) [inline rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java:73]: same here

File: rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java [hunk 6]
```diff
@@ -76,7 +84,7 @@ public int addOneOffLockWhitelistAddress(Transaction tx, String addressBase58, B
     @Override
     public int addUnlimitedLockWhitelistAddress(Transaction tx, String addressBase58) {
         try {
-            Address address = Address.fromBase58(networkParameters, addressBase58);
+            Address address = Address.fromBase58(whitelistConstants.getBtcParams(), addressBase58);
```
Comment (julia-zack) [inline rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java:87]: and here

**Review by wilmerrootstock — APPROVED** [review]
